### PR TITLE
Rails react redux router

### PR DESCRIPTION
**Pull request by tanimmahmud**
added redux to the project

### CODE REVIEW

File: app/javascript/components/RatingQuestions/ShowRatingQuestion.tsx [hunk 1]
```diff
@@ -1,57 +1,70 @@
...
-    axios.get(`/rating_questions/${this.props.match.params.id}.json`)
-      .then(response => {
-        this.setState({
-          id: response.data.id,
-          title: response.data.title,
-          url: response.data.url
-        })
-      })
+    const { questions } = this.props;
+    const questionId = this.props.match.params.id;
```
Comment (radar) [inline app/javascript/components/RatingQuestions/ShowRatingQuestion.tsx]: Do you need to fetch the question here? Could it potentially be passed in by a parent component?

File: app/javascript/reducers/reducer_questions.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { FETCH_QUESTIONS, FETCH_QUESTION, SAVE_QUESTION, EDIT_QUESTION, DELETE_QUESTION, QuestionActionTypes } from '../actions/types';
+
+const updateEditedQuestion = (questions: any, newQuestion: any) => {
```
Comment (radar) [inline app/javascript/reducers/reducer_questions.tsx]: Careful of using the `any` type here too.

File: app/javascript/components/RatingQuestions/EditRatingQuestionForm.tsx [hunk 3]
```diff
@@ -2,29 +2,37 @@ import * as React from 'react';
...
 
 class EditRatingQuestionForm extends React.Component<EditRatingQuestionFormProps, EditRatingQuestionFormState> {
     state = {
-        value: ''
+        value: '',
+        editMessage: '',
+    }
+
+    singleQuestion = (questions: any, questionId: string) => {
+      let question = questions.filter(question => question.id === questionId)
```
[the remark this answers is not in the document]
Comment (tanimmahmud) [reply app/javascript/components/RatingQuestions/EditRatingQuestionForm.tsx]: did not know js has find!!

File: app/javascript/components/RatingQuestions/EditRatingQuestionForm.tsx [hunk 4]
```diff
@@ -2,29 +2,37 @@ import * as React from 'react';
...
+import { editQuestion, fetchQuestion } from '../../actions';
 
 interface EditRatingQuestionFormProps extends RouteComponentProps<any>, React.Props<any> {
-  id: string,
-  title: string,
-  url: string,
-  children?: React.ReactNode
+  editQuestion: (questionId: string, value: string) => void,
+  fetchQuestion: (questionId: string) => void,
+  questions: any,
```
[the remark this answers is not in the document]
Comment (tanimmahmud) [reply app/javascript/components/RatingQuestions/EditRatingQuestionForm.tsx]: was bit excited about redux part so just made ts happy with any. will fix all ts types :)

File: app/javascript/components/RatingQuestions/EditRatingQuestionForm.tsx [hunk 5]
```diff
@@ -2,29 +2,37 @@ import * as React from 'react';
...
 
     componentDidMount = () => {
-      axios.get(`/rating_questions/${this.props.match.params.id}.json`)
-      .then(response => {
-        this.setState({
-          value: response.data.title
-        })
+      const { questions } = this.props;
+      const questionId = this.props.match.params.id;
+      this.props.fetchQuestion(questionId);
```
[the remark this answers is not in the document]
Comment (tanimmahmud) [reply app/javascript/components/RatingQuestions/EditRatingQuestionForm.tsx]: i am actually fetching it from redux store. in order to pass it to form I think we need to pass it from react router and i is giving me some typescript bugs. maybe you can show me on our 1-1 ?

File: app/javascript/components/RatingQuestions/EditRatingQuestionForm.tsx [hunk 6]
```diff
@@ -53,16 +59,24 @@ class EditRatingQuestionForm extends React.Component<EditRatingQuestionFormProps
                 <form onSubmit={this.handleSubmit}>
                     <h3>Title:</h3>
                     <textarea className={[styles.field, styles.editField].join(' ')} onChange={this.handleChange} name="title" value={this.state.value}/><br/>
+                    {this.state.editMessage? <h3>{this.state.editMessage}</h3> : ''}
                     <button className={[styles.button, styles.cancelButton].join(' ')}><Link to="/">Cancel</Link></button>
-                    <button className={styles.button} type="submit" value="Submit">
-                      Update Question
-                    </button>
+                    { this.state.editMessage?
+                      <button className={[styles.button, styles.cancelButton].join(' ')}><Link to="/">Back</Link></button>
```
[the remark this answers is not in the document]
Comment (tanimmahmud) [reply app/javascript/components/RatingQuestions/EditRatingQuestionForm.tsx]: somehow i was getting an error with this. but it works now :)

File: app/javascript/components/RatingQuestions/RatingQuestions.tsx [hunk 7]
```diff
@@ -1,42 +1,51 @@
...
-  questions: Question[]
+  questions: Question[],
+  fetchQuestions(): void
 }
 
 class RatingQuestions extends React.Component<RatingQuestionsProps, {}> {
-  state = {
-    questions: []
+  componentDidMount = () => {
+    if(this.props.questions.length === 0){
```
[the remark this answers is not in the document]
Comment (tanimmahmud) [reply app/javascript/components/RatingQuestions/RatingQuestions.tsx]: that's coming from redux store ```function mapStateToProps(state): any {
  return {questions: state.questions}
}```

**Comment by tanimmahmud** [comment]
[Ramping up on Rails, react, react router and redux](https://trello.com/c/laj5rA0z/1-ramping-up-on-rails-react-react-router-and-redux)

**Pull request closed without merging**